File: Calendar.Java/src/main/java/com/example/calendar_server/configuration/CalDavOptions.java
package com.example.calendar_server.configuration;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class CalDavOptions {
    public static final String KEY = "CalDav";

    @Value("${"+ KEY + ".Username}")
    public String Username;

    @Value("${"+ KEY + ".Password}")
    public String Password;

    @Value("${"+ KEY + ".UserId}")
    public String UserId;

    @Value("${"+ KEY + ".CalendarId}")
    public String CalendarId;
}

File: Calendar.CSharp/src/Calendar.Server/Configuration/CalDavOptions.cs
namespace CalendarCore.Configuration;

public class CalDavOptions
{
    public const string Key = "CalDav";
    public required string Username { get; init; }
    public required string Password { get; init; }
    public required string UserId { get; init; }
    public required string CalendarId { get; init; }
}
